build/firmware/imx8.mk

```diff
@@ -50,16 +50,27 @@ imx8_atf:
 	
 imx8_optee:
 	cd $(IMX8_REPO_ROOT)/optee_os
+ifeq ($(IMX8_TARGET),NXPEVK_iMX8M_4GB)
 	$(MAKE) PLATFORM=imx PLATFORM_FLAVOR=$(OPTEE_PLATFORM) \
      CFG_TEE_CORE_DEBUG=n CFG_TEE_CORE_LOG_LEVEL=2  CFG_UART_BASE=0x30890000 \
      CFG_RPMB_FS=y CFG_RPMB_TESTKEY=y CFG_RPMB_WRITE_KEY=y CFG_REE_FS=n  \
-     CFG_IMXCRYPT=y CFG_CORE_HEAP_SIZE=98304 \
+     CFG_IMXCRYPT=y CFG_CORE_HEAP_SIZE=131072 \
+	 CROSS_COMPILE64=$(CROSS_COMPILE) \
+	 CROSS_COMPILE= \
+
+else

```

Comment:
For imx6/7 we have a set of common optee flags, and board specific flags per board. If there is going to be a lot of imx8 variants it might make sense to create an OPTEE_COMMON_FLAGS variable, and only append the differences per board. (see common.mk).

That said with only two boards this is quite manageable for now.

Reply:
I was thinking that.  The only difference currently is removing the explicit UART definition.